src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -284,17 +292,47 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
           console.log('_keyManager was not initialized.');
         }
 
+
         this._options = this.matSelect.options;
         this._options.changes
-          .pipe(takeUntil(this._onDestroy))
-          .subscribe(() => {
+          .pipe(
+            startWith(this._options),
+            takeUntil(this._onDestroy)
+          )
+          .subscribe((optionChanges: QueryList<MatOption>) => {
+
+            // Convert the QueryList to an array
+            const options = optionChanges.toArray();
+
             const keyManager = this.matSelect._keyManager;
             if (keyManager && this.matSelect.panelOpen) {
 
               // avoid "expression has been changed" error
               setTimeout(() => {
                 // set first item active and input width
-                keyManager.setFirstItemActive();
+
+                // The Options queryList will return the input text box
+                // as the first item, we don't consider that the 'first'
+                // item as it's not a selectable option for the user.
+                const currentFirstOption = options[1];
```

Comment:
this should depend on `this.matOption`, i.e.
```suggestion
                const currentFirstOption = options[this.matOption ? 1 : 0];
```